pkg/cloudprovider/aws/instance.go

```diff
@@ -299,6 +295,16 @@ func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Ins
 					},
 				},
 			}
+
+			if amiFamily != nil {
```

Comment:
How does scheduling work if we don't know AMIFamily? Is it possible for the customer to get this wrong? Should we default to linux? Should we require that the customer sets an OS requirement if they don't set AMIFamily? Sounds like you're arguing we should just not set it and customers may get it wrong.